File: Editorial/0128_Longest_Consecutive_Sequence/C/2_Sorting.c
int cmp(const void *a, const void *b) { return (*(int *)a - *(int *)b); }
int longestConsecutive(int *nums, int numsSize) {
    if (numsSize == 0) return 0;
    qsort(nums, numsSize, sizeof(int), cmp);
    int longestStreak = 1;
    int currentStreak = 1;
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[i - 1]) {
            if (nums[i] == nums[i - 1] + 1) {
                currentStreak += 1;
            } else {
                longestStreak = longestStreak > currentStreak ? longestStreak
                                                              : currentStreak;
                currentStreak = 1;
            }
        }
    }
    return longestStreak > currentStreak ? longestStreak : currentStreak;
}

File: Editorial/0128_Longest_Consecutive_Sequence/Go/2_Sorting.go
func longestConsecutive(nums []int) int {
    if len(nums) == 0 {
        return 0
    }
    sort.Ints(nums)
    longestStreak := 1
    currentStreak := 1
    for i := 1; i < len(nums); i++ {
        if nums[i] != nums[i-1] {
            if nums[i] == nums[i-1]+1 {
                currentStreak += 1
            } else {
                longestStreak = max(longestStreak, currentStreak)
                currentStreak = 1
            }
        }
    }
    return max(longestStreak, currentStreak)
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}
